spec/book_spec.rb

```diff
@@ -0,0 +1,58 @@
+require './book'
+require './label'
+require './genre'
+require './author'
+
+RSpec.describe Book do
+  before(:each) do
+    genre = Genre.new(name: 'Vintage')
+    author = Author.new('Mike', 'Smith')
+    label = Label.new('Classic', 'Red')
+    genre2 = Genre.new(name: 'Vintage')
+    author2 = Author.new('John', 'Doe')
+    label2 = Label.new('Bestseller', 'Yellow')
+
+    @book1 = Book.new(publisher: 'Amazon', cover_state: 'Good', genre: genre, author: author,
+                      source: 'Bookstore', label: label, publish_date: '1982-03-11')
+
+    @book2 = Book.new(
+      publisher: 'Amazon', cover_state: 'Good', genre: genre2, author: author2,
+      source: 'Bookstore', label: label2, publish_date: '1982-03-11'
+    )
+  end
+
+  it 'lists all labels' do
+    Label.new('Classic', 'grey')
+    Label.new('Bestseller', 'red')
+
+    expect(Book.all.empty?).to eq(false)
+    expect(Label.all.length).not_to eq(0)
+  end
+
+  it 'checks if book can be archived' do
+    expect(@book1.archived).to eq(false)
+    expect(@book2.archived).to eq(false)
+  end
+end
+
+RSpec.describe Book do
+  genre = Genre.new(name: 'Vintage')
+  author = Author.new('Mike', 'Smith')
+  label = Label.new('Classic', 'Red')
+  genre2 = Genre.new(name: 'Vintage')
+  author2 = Author.new('John', 'Doe')
+  label2 = Label.new('Bestseller', 'Yellow')
+
+  @book1 = Book.new(publisher: 'Amazon', cover_state: 'Good', genre: genre, author: author,
+                    source: 'Bookstore', label: label, publish_date: '1982-03-11')
+
+  @book2 = Book.new(
+    publisher: 'Amazon', cover_state: 'Good', genre: genre2, author: author2,
+    source: 'Bookstore', label: label2, publish_date: '1982-03-11'
+  )
+
+  it 'lists all books' do
+    # checks that books are created before each test
+    expect(Book.all.length).to eq(6)
+  end
```

Comment:
- [ ] Kindly fix this failing test. 

<img width="465" alt="Untitled" src="https://github.com/Victoria129/ruby-capstone/assets/103386183/fbad33f5-39c0-4b43-8f41-2d690a6a5c60">

Reply:
I will, thank you.